ddtrace/propagation/http.py

```diff
@@ -99,6 +99,9 @@ def my_controller(url, headers):
                 normalized_headers,
                 default=0,
             )
+            if not trace_id:
```

Comment:
```suggestion
            if trace_id is not None:
```

trace_id of 0 is technically valid I think 🤔

Reply:
not sure the use case where this would make sense.

Sending context propagation headers for sampling and origin but no tracing data? or a parent id, but not a trace id?

if the trace id doesn't exist then the other headers just don't make sense to keep.

trace id with a parent id of 0 makes sense (at least when used with origin)

Reply:
also, the trace id default is `0`, so would the `is not None` case even work?

Reply:
Oh wow yeah the default is 0 🤔. Ok I'm fine with leaving this as-is then, probably something in the future to consider

Reply:
we could make the default `None`, but I still don't think `0` is a valid value, and in fact should invalidate the entire propagated context

Reply:
Hmm do we have that specified anywhere? I know Java uses `0` as a parent_id to represent root spans. I suppose they would probably also use `0` as "null" traces?

Reply:
`0` for parent id is fine, but not for a trace id. but again, why should we support propagating a null trace?

Reply:
My point here was whether trace_id of 0 is null or not and hence whether we're treating it correctly.

Reply:
oh... this gets interesting, since if it exists but is "0", then `trace_id` will be `"0"`, if it isn't provided it'll be the default of `0`, so... we aren't even properly checking for `"0"` being sent, `0` is just a proxy for "not provided".

🤦🏻

Reply:
@P403n1x87 we probably want to leave a comment here  about ☝🏻 

or we might want to clarify the code a little more (like make the default be `None` not `0`)

since, if the trace id is provided but `"0"` it'll pass this check and continue processing.

where really this check ends up being "X-Datadog-Trace-Id was not provided"

which changing to `default=None` and this check as `if trace_id is not None` will make that more clear?

what do you think?

Reply:
Agreed. I like the idea of using `None` for "header not set".